tvloader/parser2v1/parse_file_test.go

```diff
@@ -885,4 +886,58 @@ func TestParser2_1FailsIfArtifactURIBeforeArtifactName(t *testing.T) {
 	}
 }
 
+func TestParser2_1FilesWithoutSpdxIdThrowError(t *testing.T) {
+	// case 1
+	// Last unpackaged file no packages in doc
+	// Last file of last package in the doc
+	tvPairs := []reader.TagValuePair{
+		{Tag: "SPDXVersion", Value: "SPDX-2.1"},
+		{Tag: "DataLicense", Value: "CC0-1.0"},
+		{Tag: "SPDXID", Value: "SPDXRef-DOCUMENT"},
+		{Tag: "FileName", Value: "f1"},
+	}
+	_, err := ParseTagValues(tvPairs)
+	if err == nil {
+		t.Errorf("files withoutSpdx Identifiers getting accepted")
+	}
+
+	// case 2 : The previous file (packaged or unpackaged does not contain spdxID)
```

Comment:
I think case 2 makes sense to keep here. However, I would not structure it using `ParseTagValues()` and a `[]reader.TagValuePair`. Instead, I would do it the way you did for cases 3 and 4 below -- create a `tvParser2_1{}` state object with the start state, and then pass the tag and value to `parser.parsePair2_1()`.

The reason for this is that we want these to be unit tests -- to each be testing as little functionality as possible. If we use ParseTagValues() and `reader.TagValuePair{}`, we have to import the reader package into these tests, and we have broader functionality running in this test than we need.